test/integration/fleetscope/fleetscope_test.go

```diff
@@ -41,11 +44,94 @@ func TestFleetscope(t *testing.T) {
 				"fleet_project_id":       multitenant.GetStringOutput("fleet_project_id"),
 				"cluster_membership_ids": multitenant.GetStringOutputList("cluster_membership_ids"),
 			}
+
 			fleetscope := tft.NewTFBlueprintTest(t,
 				tft.WithTFDir(fmt.Sprintf("../../../4-fleetscope/envs/%s", envName)),
 				tft.WithVars(vars),
 				tft.WithRetryableTerraformErrors(testutils.RetryableTransientErrors, 3, 2*time.Minute),
 			)
+
+			fleetscope.DefineVerify(func(assert *assert.Assertions) {
+				fleetscope.DefaultVerify(assert)
+
+				// Outputs
+				fleetProjectID := multitenant.GetStringOutput("fleet_project_id")
+				region := multitenant.GetStringOutputList("cluster_regions")[0]
+
+				// Service Account
+				rootReconcilerSa := fmt.Sprintf("root-reconciler@%s.iam.gserviceaccount.com", fleetProjectID)
+
+				iamFilter := fmt.Sprintf("bindings.members:'serviceAccount:%s'", rootReconcilerSa)
+				iamCommonArgs := gcloud.WithCommonArgs([]string{"--flatten", "bindings", "--filter", iamFilter, "--format", "json"})
+				projectPolicyOp := gcloud.Run(t, fmt.Sprintf("projects get-iam-policy %s", fleetProjectID), iamCommonArgs).Array()
+				saListRoles := testutils.GetResultFieldStrSlice(projectPolicyOp, "bindings.role")
+				assert.Subset(saListRoles, []string{"roles/source.reader"}, fmt.Sprintf("service account %s should have \"roles/source.reader\" project level role", rootReconcilerSa))
+
+				saPolicyOp := gcloud.Runf(t, "iam service-accounts get-iam-policy %s --project %s", rootReconcilerSa, fleetProjectID)
```

Comment:
```suggestion
				saPolicyOp := gcloud.Runf(t, "iam service-accounts get-iam-policy %s --project %s", rootReconcilerSa, fleetProjectID).Array()[0]
```

I believe that this changes would avoid the need to access the index inside the get methods in multiple places (i.e `saPolicyOp.Get("bindings.0.role")`).

My only concern would be to make sure this command always return a single record, otherwise we could add `--sort-by` or `--filter` to make sure the expected record is in the 0 index always.

Reply:
Just tried using **flatten** and **filter** bellow and had the same result:

_Command:_ `gcloud iam service-accounts get-iam-policy root-reconciler@{PROJECT_ID}.iam.gserviceaccount.com --project {PROJECT_ID} --flatten bindings --filter bindings.role:"roles/iam.workloadIdentityUser" --format json`

_Result:_
```
[
  {
    "bindings": {
      "members": [
        "serviceAccount:{PROJECT_ID}.svc.id.goog[config-management-system/root-reconciler]"
      ],
      "role": "roles/iam.workloadIdentityUser"
    },
    "etag": "BwYUK3XQiNs=",
    "version": 1
  }
]
```

Reply:
adding the filter would force it to have a single value,
 the one for the `"role": "roles/iam.workloadIdentityUser"`.
without the filter, a change in the code, for example granting more roles could make the test get the wrong role in the response.
it is a coincidence that the policy has only one role